catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/NfcTagListFragment.java

```diff
@@ -105,7 +106,17 @@ public void onNewIntent(Intent intent) {
 					.getUniqueNameInNameables(getString(R.string.default_tag_name), adapter.getItems());
 			item.setName(name);
 			item.setNfcTagUid(uid);
-			adapter.add(item);
+			Iterator<NfcTagData> it = adapter.getItems().iterator();
```

Comment:
Instead of using an Iterator here I think it would be better to override equals() and hashCode() of NfcTagData and then simply call adapter.getItems().contains() to check whether its a duplicate item or not.